src/nuclide_header.F90

```diff
@@ -0,0 +1,761 @@
+      procedure, pass :: get_xs => nuclideiso_get_xs ! Gets Size of Data w/in Object
+      procedure, pass :: calc_f => nuclideiso_calc_f ! Calcs f given mu
+  end type NuclideIso
+
+!===============================================================================
+! NuclideAngle contains the base MGXS data for a nuclide specifically for
+! explicit angle-dependent weighted MGXS
+!===============================================================================
+
+  type, extends(NuclideMG) :: NuclideAngle
+
+    ! Microscopic cross sections. Dimensions are: (Npol, Nazi, Nl, Ng, Ng)
+    real(8), allocatable :: total(:,:,:)        ! total cross section
+    real(8), allocatable :: absorption(:,:,:)   ! absorption cross section
+    real(8), allocatable :: scatter(:,:,:,:,:)  ! scattering information
+    real(8), allocatable :: nu_fission(:,:,:,:) ! fission matrix (Gout x Gin)
+    real(8), allocatable :: k_fission(:,:,:)    ! kappa-fission
+    real(8), allocatable :: fission(:,:,:)      ! neutron production
+    real(8), allocatable :: chi(:,:,:)          ! Fission Spectra
+    real(8), allocatable :: mult(:,:,:,:)       ! Scatter multiplicity (Gout x Gin)
+
+    ! In all cases, right-most indices are theta, phi
+    integer              :: Npol         ! Number of polar angles
+    integer              :: Nazi         ! Number of azimuthal angles
+    real(8), allocatable :: polar(:)     ! polar angles
+    real(8), allocatable :: azimuthal(:) ! azimuthal angles
+
+    ! Type-Bound procedures
+    contains
+      procedure, pass :: print  => nuclideangle_print  ! Gets Size of Data w/in Object
+      procedure, pass :: get_xs => nuclideangle_get_xs ! Gets Size of Data w/in Object
+      procedure, pass :: calc_f => nuclideangle_calc_f ! Calcs f given mu
+  end type NuclideAngle
+
+!===============================================================================
+! NUCLIDEMGCONTAINER pointer array for storing Nuclides
+!===============================================================================
+
+  type NuclideMGContainer
+    class(NuclideMG), pointer :: obj
+  end type NuclideMGContainer
+
+!===============================================================================
+! NUCLIDE0K temporarily contains all 0K cross section data and other parameters
+! needed to treat resonance scattering before transferring them to NuclideCE
+!===============================================================================
+
+  type Nuclide0K
+
+    character(10) :: nuclide             ! name of nuclide, e.g. U-238
+    character(16) :: scheme = 'ares'     ! target velocity sampling scheme
+    character(10) :: name                ! name of nuclide, e.g. 92235.03c
+    character(10) :: name_0K             ! name of 0K nuclide, e.g. 92235.00c
+    real(8)       :: E_min = 0.01e-6_8   ! lower cutoff energy for res scattering
+    real(8)       :: E_max = 1000.0e-6_8 ! upper cutoff energy for res scattering
+
```

Comment:
I'd get rid of the blank lines before and after the variable declarations to match other derived type blocks.